### PR TITLE
Transformer rework

### PR DESCRIPTION
**Pull request by Fireboltofdeath**
This drastically improves performance of transformers for subsequent compilations in watch mode. There is still a large initial cost for using transformers however this should make them far more tolerable when used with watch mode.

Before:
(3 transformers)
image
(1 transformer)
image

After:
(3 transformers)
image
(1 transformer)
image

### CODE REVIEW

File: src/Project/transformers/createTransformerWatcher.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+function createServiceHost(program: ts.Program) {
+	const rootFileNames = program.getRootFileNames().map(x => x);
+	const files = new Map<string, number>();
+
+	rootFileNames.forEach(fileName => {
+		files.set(fileName, 0);
+	});
+
+	const overridedText = new Map<string, string>();
```
Comment (osyrisrblx) [inline src/Project/transformers/createTransformerWatcher.ts:12]: `overriddenText`?

File: src/Project/transformers/createTransformerWatcher.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+	function updateFile(fileName: string, text: string) {
+		overridedText.set(fileName, text.length ? text : "\0");
```
Comment (osyrisrblx) [inline src/Project/transformers/createTransformerWatcher.ts:15]: Is the `\0` important here? If it is, can you add a comment explaining why?

**Review by osyrisrblx — APPROVED** [review]

File: src/Project/transformers/createTransformerWatcher.ts (hunk 2 above)
Comment (Fireboltofdeath) [reply src/Project/transformers/createTransformerWatcher.ts:15]: The \0 is necessary here, I'll add a comment explaining it

Comment (Fireboltofdeath) [reply src/Project/transformers/createTransformerWatcher.ts:15]: Actually, it's not necessary here. I'll remove it.

**Pull request merged**